### PR TITLE
4.0.0

### PR DESCRIPTION
**Pull request by skjnldsv**
## [v4.0.0](https://github.com/nextcloud/nextcloud-vue/tree/v4.0.0) (2021-06-04)

[Full Changelog](https://github.com/nextcloud/nextcloud-vue/compare/v3.10.1...v4.0.0)

## Breaking

- Remove deprecated legacy avatar user status [\#1945](https://github.com/nextcloud/nextcloud-vue/pull/1945) ([skjnldsv](https://github.com/skjnldsv))
- Fix vue material icons [\#1816](https://github.com/nextcloud/nextcloud-vue/pull/1816) ([skjnldsv](https://github.com/skjnldsv))
- Resizable columns [\#1759](https://github.com/nextcloud/nextcloud-vue/pull/1759) ([brueckner](https://github.com/brueckner))
- Bump @nextcloud/browserslist-config from 1.0.0 to 2.1.0 [\#1786](https://github.com/nextcloud/nextcloud-vue/pull/1786) ([dependabot](https://github.com/dependabot))

## Enhancements

- Add CheckboxRadio component [\#1875](https://github.com/nextcloud/nextcloud-vue/pull/1875) ([skjnldsv](https://github.com/skjnldsv))
- Add disabled state to RichContenteditable and clean not-working v-model [\#1894](https://github.com/nextcloud/nextcloud-vue/pull/1894) ([skjnldsv](https://github.com/skjnldsv))
- Add loading state to CheckboxRadio [\#1889](https://github.com/nextcloud/nextcloud-vue/pull/1889) ([skjnldsv](https://github.com/skjnldsv))
- Add proper opening/opened/closing/closed events & fix events docs [\#1914](https://github.com/nextcloud/nextcloud-vue/pull/1914) ([skjnldsv](https://github.com/skjnldsv))
- Add slot for third line in listitem component [\#1888](https://github.com/nextcloud/nextcloud-vue/pull/1888) ([ma12-co](https://github.com/ma12-co))
- Allow material design icon in breadcrumb [\#1938](https://github.com/nextcloud/nextcloud-vue/pull/1938) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Allow material design icons in AppSidebarTab [\#1939](https://github.com/nextcloud/nextcloud-vue/pull/1939) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Create ActionCaption component [\#1739](https://github.com/nextcloud/nextcloud-vue/pull/1739) ([ma12-co](https://github.com/ma12-co))
- Create ListItem component [\#616](https://github.com/nextcloud/nextcloud-vue/pull/616) ([ma12-co](https://github.com/ma12-co))
- Create ListItem component [\#616](https://github.com/nextcloud/nextcloud-vue/pull/616) ([ma12-co](https://github.com/ma12-co))
- Fix alignment of md icons in single action [\#1886](https://github.com/nextcloud/nextcloud-vue/pull/1886) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Implement switch mode to CheckboxRadioSwitch [\#1890](https://github.com/nextcloud/nextcloud-vue/pull/1890) ([skjnldsv](https://github.com/skjnldsv))
- Improve documentation sections [\#1883](https://github.com/nextcloud/nextcloud-vue/pull/1883) ([skjnldsv](https://github.com/skjnldsv))
- Make avatar menu focusable with keyboard [\#1854](https://github.com/nextcloud/nextcloud-vue/pull/1854) ([PVince81](https://github.com/PVince81))
- Replace AppNavigationCounter with CounterBubble [\#1895](https://github.com/nextcloud/nextcloud-vue/pull/1895) ([PVince81](https://github.com/PVince81))
- Update appnavigationcaption [\#1863](https://github.com/nextcloud/nextcloud-vue/pull/1863) ([ma12-co](https://github.com/ma12-co))
- Wrap icon span in icon slot for ActionButton [\#1930](https://github.com/nextcloud/nextcloud-vue/pull/1930) ([raimund-schluessler](https://github.com/raimund-schluessler))

## Fixed bugs

- Add id prop to ListItemIcon to prevent v-binding to the Avatar [\#1833](https://github.com/nextcloud/nextcloud-vue/pull/1833) ([skjnldsv](https://github.com/skjnldsv))
- Allow to append datepicker to body [\#1880](https://github.com/nextcloud/nextcloud-vue/pull/1880) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Correctly align md icon in ActionRouter component [\#1927](https://github.com/nextcloud/nextcloud-vue/pull/1927) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Fix Actions css [\#1834](https://github.com/nextcloud/nextcloud-vue/pull/1834) ([jotoeri](https://github.com/jotoeri))
- Fix appnavigationitem layout [\#1897](https://github.com/nextcloud/nextcloud-vue/pull/1897) ([ma12-co](https://github.com/ma12-co))
- Fix counter margin [\#1912](https://github.com/nextcloud/nextcloud-vue/pull/1912) ([ma12-co](https://github.com/ma12-co))
- Fix datepicker overflow when week numbers are shown [\#1917](https://github.com/nextcloud/nextcloud-vue/pull/1917) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Fix first and last action input padding [\#1967](https://github.com/nextcloud/nextcloud-vue/pull/1967) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Fix first-action icon class not showing when Actions icon-slot is def… [\#1879](https://github.com/nextcloud/nextcloud-vue/pull/1879) ([nickvergessen](https://github.com/nickvergessen))
- Fix ListItemIcon listeners [\#1870](https://github.com/nextcloud/nextcloud-vue/pull/1870) ([skjnldsv](https://github.com/skjnldsv))
- Fix md icon size in ActionInput [\#1966](https://github.com/nextcloud/nextcloud-vue/pull/1966) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Fix user bubble when username contains a space [\#1948](https://github.com/nextcloud/nextcloud-vue/pull/1948) ([artonge](https://github.com/artonge))
- Prevent overflow of action-link in Firefox [\#1934](https://github.com/nextcloud/nextcloud-vue/pull/1934) ([raimund-schluessler](https://github.com/raimund-schluessler))
- Remove AppNavigationItem flex order for AppNavigationCaption [\#1911](https://github.com/nextcloud/nextcloud-vue/pull/1911) ([skjnldsv](https://github.com/skjnldsv))
- Use ActionButton for AppNavigationToggle [\#1844](https://github.com/nextcloud/nextcloud-vue/pull/1844) ([raimund-schluessler](https://github.com/raimund-schluessler))

### CODE REVIEW

**Comment by skjnldsv** [comment]
Damnit, the changelog generator is failing

**Review by GretaD — APPROVED** [review]

**Review by korelstar — DISMISSED** [review]
I found two design issues in 4.0.0 (see screenshots below):
- `AppNavigationCaption` is aligned right. According to https://github.com/nextcloud/nextcloud-vue/pull/1863, this is not correct.
- `AppNavigationItem`'s edit mode has no vissible cancel button.

## nc-vue 3.10.1
Screenshot before

## nc-vue 4.0.0
Screenshot after

**Comment by skjnldsv** [comment]
@ma12-co ?

**Comment by ma12-co** [comment]
@korelstar the component looks ok in the documentation, are you sure this is not due to custom css you have in your app? I'll test this with talk too

**Review by ChristophWurst — APPROVED** [review]

**Comment by skjnldsv** [comment]
Rebased, let's see documentation

**Comment by ma12-co** [comment]
> Rebased, let's see documentation

Looks good

**Review by ma12-co — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,54 @@
 # Changelog
 
 All notable changes to this project will be documented in this file.
+## [v4.0.0](https://github.com/nextcloud/nextcloud-vue/tree/v4.0.0) (2021-06-04)
```
Comment (PVince81) [inline CHANGELOG.md:4]: change to 2021-06-07 before merging ?

Comment (skjnldsv) [reply CHANGELOG.md:4]: Yes

**Comment by korelstar** [comment]
> @korelstar the component looks ok in the documentation, are you sure this is not due to custom css you have in your app? I'll test this with talk too

You can reproduce the first issue (`AppNavigationCaption` is aligned right) in the docs by removing the `ActionButton` from the example.

You can reproduce the second issue (editable `AppNavigationItem`) in the docs by changing the width of the `app-navigation-entry` container to `300px` like in a real app-navigation.

Did you ever tested the changes in a real app?

**Comment by skjnldsv** [comment]
> Did you ever tested the changes in a real app?

@korelstar 
Are you speaking globally about this pr or just this specific use-case?

**Comment by korelstar** [comment]
I was just wondering about the reaction that everything is looking fine. 😉

**Comment by skjnldsv** [comment]
Not everyone uses edit in the appNavigation

**Comment by korelstar** [comment]
Sorry, that was not meant as affront. Maybe we can surround all `AppNavigationItem` examples in the docs with `<ul style="width: 300px">`? That would make the examples more realistic.

**Comment by skjnldsv** [comment]
> Sorry, that was not meant as affront

Not taken as is :) :hugs:

**Comment by skjnldsv** [comment]
Seems ok to me, LGTM

300px: 
image

Normal on docs:
image

**Comment by skjnldsv** [comment]
image
image

**Comment by skjnldsv** [comment]
Caption fix https://github.com/nextcloud/nextcloud-vue/pull/2024

**Pull request merged**